From e99055fcab52f0756739c264355ef89225d03070 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 28 Jun 2016 18:04:24 +0300
Subject: [PATCH 0304/1345] net: mvpp2x: gop: remove unused gop debug

commit  e672416fda276e320f1656e758b48f9a4463509e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- GOP print debug moved out of else state
- remove unused GOP flags
- remove unused GOP debug

Change-Id: I2896947eafd3bdb0259e4b6ed2e0e7fce3866a4e
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30799
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c |   11 +----------
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.h |   13 -------------
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h      |    3 ---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |   11 ++---------
 4 files changed, 3 insertions(+), 35 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
index dba925a..8cd8d37 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
@@ -978,8 +978,6 @@ int mv_gop110_port_init(struct gop_hw *gop, struct mv_mac_data *mac)
 		return -1;
 	}
 
-	GOP_DEBUG(gop->gop_port_debug.flags = (1 << CREATED));
-
 	return 0;
 }
 
@@ -1069,8 +1067,6 @@ void mv_gop110_port_enable(struct gop_hw *gop, struct mv_mac_data *mac)
 		pr_err("%s: Wrong port mode (%d)", __func__, mac->phy_mode);
 		return;
 	}
-
-	GOP_DEBUG(gop->gop_port_debug[port_num].flags |= (1 << ENABLED));
 }
 
 void mv_gop110_port_disable(struct gop_hw *gop, struct mv_mac_data *mac)
@@ -1094,8 +1090,6 @@ void mv_gop110_port_disable(struct gop_hw *gop, struct mv_mac_data *mac)
 		pr_err("%s: Wrong port mode (%d)", __func__, mac->phy_mode);
 		return;
 	}
-
-	GOP_DEBUG(gop->gop_port_debug[port_num].flags &= ~(1 << ENABLED));
 }
 
 void mv_gop110_port_periodic_xon_set(struct gop_hw *gop,
@@ -1312,10 +1306,7 @@ int mv_gop110_status_show(struct gop_hw *gop, struct mv_mac_data *mac)
 	bool port_en = false;
 
 	mv_gop110_port_link_status(gop, mac, &port_status);
-#ifdef MV_PP22_GOP_DEBUG
-	port_en = (gop->gop_port_debug[port_num].flags &
-		  (1 << ENABLED)) ? true : false;
-#endif
+
 	pr_info("-------------- Port %d configuration ----------------",
 		port_num);
 
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.h b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.h
index fda839c..672de43 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.h
@@ -99,19 +99,6 @@ enum mv_netc_lanes {
 	MV_NETC_LANE_45,
 };
 
-/* pp3_gop_ctrl flags */
-#define mv_gop_F_DEBUG_BIT		0
-#define mv_gop_F_ATTACH_BIT		1
-
-#define mv_gop_F_DEBUG		(1 << mv_gop_F_DEBUG_BIT)
-#define mv_gop_F_ATTACH		(1 << mv_gop_F_ATTACH_BIT)
-
-enum gop_port_flags {NOT_CREATED, CREATED, UNDER_RESET, ENABLED};
-
-struct gop_port_ctrl {
-	u32  flags;
-};
-
 #define MV_RGMII_TX_FIFO_MIN_TH		(0x41)
 #define MV_SGMII_TX_FIFO_MIN_TH		(0x5)
 #define MV_SGMII2_5_TX_FIFO_MIN_TH	(0xB)
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index 97bfa91..c86150f 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -423,9 +423,6 @@ struct cpn110_gop_hw {
 	void __iomem *xpcs_base;
 	void __iomem *rfu1_base;
 
-#ifdef MV_PP22_GOP_DEBUG
-	static struct gop_port_ctrl gop_port_debug[MVCPN110_GOP_MAC_NUM];
-#endif
 };
 
 struct gop_hw {
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index efd1fa4..42cd271 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -3630,9 +3630,6 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 	} else {
 		port->mac_data.force_link = false;
 
-		pr_debug("gop_mac(%d), phy_mode(%d) (%s)\n", id,  phy_mode,
-			phy_modes(phy_mode));
-
 		switch (phy_mode) {
 		case PHY_INTERFACE_MODE_SGMII:
 			speed = 0;
@@ -3665,6 +3662,8 @@ static int mv_pp2_init_emac_data(struct mv_pp2x_port *port,
 		}
 	}
 	port->mac_data.phy_mode = phy_mode;
+	pr_debug("gop_mac(%d), phy_mode(%d) (%s)\n", id,  phy_mode,
+		phy_modes(phy_mode));
 	pr_debug("gop_mac(%d), phy_speed(%d)\n", id,  port->mac_data.speed);
 
 
@@ -4930,12 +4929,6 @@ static int mv_pp2x_probe(struct platform_device *pdev)
 		goto err_clk;
 	}
 
-	/*Init GOP */
-
-#ifdef MV_PP22_GOP_DEBUG
-	for (i = 0; i < MVCPN110_GOP_MAC_NUM; i++)
-		hw->gop.gop_110.gop_port_debug[i].flags = (1 << NOT_CREATED);
-#endif
 #if !defined(CONFIG_MV_PP2_FPGA) && !defined(CONFIG_MV_PP2_PALLADIUM)
 	/* smi init */
 	if (priv->pp2_version == PPV21)
-- 
1.7.9.5

